File: output/py/Medium/611-longest-substring-without-repeating-characters.py
#https://leetcode.com/problems/longest-substring-without-repeating-characters/
#Given a string s, find the length of the longest substring without repeating characters.


class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """       
        res = 0
        for i in range(len(s)):
            for j in range(i+1,len(s)+1):
                if len(set(s[i:j])) == j-i:
                    res = max(res,j-i)
        return res
        

    

File: output/c/Medium/611-longest-substring-without-repeating-characters.c
/**https://leetcode.com/problems/longest-substring-without-repeating-characters/ */
//Given a string s, find the length of the longest substring without repeating characters.


int lengthOfLongestSubstring(char * s){
    int i = 0;
    int j = 0;
    int k = 0;
    int max = 0;
    int count = 0;
    int len = strlen(s);
    int flag = 0;
    for(i = 0; i < len; i++){
        for(j = i; j < len; j++){
            for(k = i; k < j; k++){
                if(s[k] == s[j]){
                    flag = 1;
                    break;
                }
            }
            if(flag == 1){
                flag = 0;
                break;
            }
            count++;
        }
        if(count > max){
            max = count;
        }
        count = 0;
    }
    return max;

}

    